make/targets/golang/verify-update.mk

```diff
@@ -28,6 +28,29 @@ verify-golint:
 	$(GOLINT) $(GO_PACKAGES)
 .PHONY: verify-govet
 
+verify-gomod:
+	$(info Running `$(GO) mod tidy`.)
+	@if ! [ -f go.mod ]; then \
+		echo "$@ failed - please run \`go mod init\`"; \
+		exit 1; \
+	fi; \
+	if ! [ -f go.sum ]; then \
+		echo "$@ failed - please run \`make update-gomod\`"; \
+		exit 1; \
+	fi; \
+	TMP_GOMOD=$$( mktemp ); \
+	TMP_GOSUM=$$( mktemp ); \
+	cp --preserve=all go.mod $${TMP_GOMOD}; \
+	cp --preserve=all go.sum $${TMP_GOSUM}; \
+	$(GO) mod tidy; \
```

Comment:
is this working without internet? Does it have to? @tnozicka @jupierce

Reply:
ping @tnozicka @jupierce

Reply:
I can at least verify that connectivity to servers outside of the Red Hat firewall will be impossible during downstream/production builds. So if `tidy` does reach out, it will fail.

Reply:
There are 2 classes of jobs:
1. Run only in CI.
Some of these might by nature require internet connection, but they are not essential so ART is free to drop them in an offline build. The subset is usually targeted at devs. Like verify-deps. If GH goes down and we have to ship an emergency fix in the meantime and we by accident break verify-deps, that's fine and we will fix it the next day.
2. Run also by ART
Build, verify, unit test - has to work offline.

Reply:
also a reason why `verify-deps` is not hooked into `verify` but it is a separate target / job